bedrock/3_EntityWorkflow/workflows.py

```diff
@@ -0,0 +1,174 @@
+import asyncio
+from datetime import timedelta
+from temporalio import workflow
+from collections import deque
+from typing import List, Tuple, Deque, Optional
+from dataclasses import dataclass
+
+with workflow.unsafe.imports_passed_through():
+    from activities import prompt_bedrock
+
+
+@dataclass
+class BedrockParams:
+    conversation_summary: Optional[str] = None
+    prompt_queue: Optional[Deque[str]] = None
+
+
+@workflow.defn
+class EntityBedrockWorkflow:
+    def __init__(self) -> None:
+        # List to store prompt history
+        self.conversation_history: List[Tuple[str, str]] = (
+            []
+        )
+        self.prompt_queue: Deque[str] = deque()
+        self.conversation_summary: Optional[str] = None
+        self.continue_as_new_per_turns: int = 6
+        self.chat_ended: bool = False
+
+    @workflow.run
+    async def run(
+        self,
+        params: BedrockParams = None,
+    ) -> str:
+
+        if params and params.conversation_summary:
+            self.conversation_history.append(
+                ("conversation_summary", params.conversation_summary)
+            )
+
+        if params and params.prompt_queue:
+            for prompt in params.prompt_queue:
+                self.prompt_queue.append(prompt)
+
+        summary_activity_task: Optional[asyncio.Task] = None
+
+        while True:
+            workflow.logger.info("Waiting for prompts...")
+
+            # Wait for a chat message (signal) or timeout
+            await workflow.wait_condition(
+                lambda: bool(self.prompt_queue) or self.chat_ended
+            )
+
+            # if end chat signal was sent
+            if self.chat_ended:
+                # the workflow might be continued as new without any
+                # chat to summarize
+                if summary_activity_task is not None:
+                    # ensure conversation summary task has finished
+                    # before closing the workflow (avoid race)
+                    await workflow.wait_condition(
+                        lambda: summary_activity_task.done()
+                    )
```

Comment:
This can cause signal loss if a signal comes in while waiting on this summary

Reply:
@cretz when you say signal loss, do you mean this code would behave like so:
* workflow is waiting on summary before continuing as new
* a signalABC comes in while waiting on summary
* summary finishes and workflow does CAN without handling signalABC (signal is effectively dropped)

How does one manage this in Python? I know in Go that we can explicitly drain signals but in the Python docs I don't see anything about draining/signal loss (and thought that Rust Core SDKs ones don't need this).

Reply:
> How does one manage this in Python? I know in Go that we can explicitly drain signals but in the Python docs I don't see anything about draining/signal loss

You do similar to Go. You don't need explicit "draining" in Python because we guarantee your handler code is _always_ called which we don't in Go. But from a business-logic POV, don't CAN while there are signals to be processed.